scripts/create-aks-acr.sh

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+: "${AKS_NAME:?Environment variable empty or not defined.}"
+: "${ACR_NAME:?Environment variable empty or not defined.}"
+
+register_feature() {
+  # pinning to 0.5.87 because of https://github.com/Azure/azure-cli/issues/23267
+  az extension add --name aks-preview --version 0.5.87
+  # register enable oidc preview feature
+  az feature register --namespace Microsoft.ContainerService --name EnableOIDCIssuerPreview >/dev/null
+  # https://docs.microsoft.com/en-us/azure/aks/windows-container-cli#add-a-windows-server-node-pool-with-containerd-preview
+  az feature register --namespace Microsoft.ContainerService --name UseCustomizedWindowsContainerRuntime >/dev/null
+  while [[ "$(az feature list --query "[?contains(name, 'Microsoft.ContainerService/EnableOIDCIssuerPreview')].{Name:name,State:properties.state}" | jq -r '.[].State')" != "Registered" ]] &&
+    [[ "$(az feature list --query "[?contains(name, 'Microsoft.ContainerService/UseCustomizedWindowsContainerRuntime')].{Name:name,State:properties.state}" | jq -r '.[].State')" != "Registered" ]]; do
+    sleep 20
+  done
+  az provider register --namespace Microsoft.ContainerService
+}
+
+main() {
+  export -f register_feature
+  # might take around 20 minutes to register
+  timeout --foreground 1200 bash -c register_feature
+  # get the latest patch version of 1.24
+  KUBERNETES_VERSION="$(az aks get-versions --location "${LOCATION}" --query 'orchestrators[*].orchestratorVersion' -otsv | grep '1.24' | tail -1)"
```

Comment:
If `KUBERNETES_VERSION` is determined here, what is the significance of the `KUBERNETES_VERSION` defined in the gh workflow?

Reply:
Right, it's removed.